.github/workflows/ci.yaml

```diff
@@ -38,21 +39,33 @@ jobs:
           git clone --depth 1 https://github.com/pulp/pulp_file.git
           git clone --depth 1 https://github.com/pulp/pulp-openapi-generator.git
 
-      - name: Install podman compose
-        if: ${{ matrix.env.TEST == 'podman' }}
+      - name: (Ubuntu) Install podman compose
+        if: matrix.env.TEST == 'podman' && matrix.os == 'ubuntu-latest'
         run: |
           sudo apt update
           sudo apt install -y httpie podman
           sudo pip install podman-compose
           podman --version
-  
-      - name: Install docker compose
-        if: ${{ matrix.env.TEST == 'docker' }}
+
+      - name: (Ubuntu) Install docker compose
+        if: matrix.env.TEST == 'docker' && matrix.os == 'ubuntu-latest'
         run: |
           sudo apt update
           sudo curl -fsSL https://get.docker.com -o get-docker.sh
           sudo sh ./get-docker.sh
 
+      - name: (Mac) Install podman compose
+        if: matrix.env.TEST == 'podman' && matrix.os == 'macos-latest'
+        run: |
+          brew install podman
+          sudo pip install httpie podman-compose
+          podman --version
+
+      - name: (Mac) Install docker compose
+        if: matrix.env.TEST == 'docker' && matrix.os == 'macos-latest'
+        run: |
+          brew install docker-compose
```

Comment:
I have no idea on how to make it work on mac

Reply:
When I get back home, I can setup remote access to my Macbook.

You could also try using a tmate debug shell:
https://github.com/marketplace/actions/debugging-with-tmate